vllm/v1/structured_output/__init__.py

```diff
@@ -36,10 +36,10 @@ def __init__(self, vllm_config: VllmConfig, max_cache_size: int = 500):
             parallel_config=vllm_config.parallel_config,
             lora_config=vllm_config.lora_config)  # type: ignore[arg-type]
         tokenizer_group.ping()
-        self.vocab_size = vllm_config.model_config.get_vocab_size()
         self.vllm_config = vllm_config
 
         tokenizer = tokenizer_group.get_lora_tokenizer(None)
+        self.vocab_size = len(tokenizer.get_vocab())
```

Comment:
Could use `len(tokenizer)`, which is cached: https://github.com/vllm-project/vllm/blob/main/vllm/transformers_utils/tokenizer.py#L101

Or it might be better to use `tokenizer.max_token_id`. I have seen cases where the vocab size is actually larger than the number of tokens in the tokenizer since some were removed.